src/core/usubscription.rs

```diff
@@ -26,6 +26,41 @@ pub use crate::up_core_api::usubscription::{
 
 use crate::{UStatus, UUri};
 
+// Tracks information for the SubscriptionCache
+pub struct SubscriptionInformation {
+    pub topic: UUri,
+    pub subscriber: SubscriberInfo,
+    pub status: SubscriptionStatus,
+    pub attributes: SubscribeAttributes,
+    pub config: EventDeliveryConfig,
+}
+
+impl Eq for SubscriptionInformation {}
+
+impl PartialEq for SubscriptionInformation {
+    fn eq(&self, other: &Self) -> bool {
+        self.subscriber == other.subscriber
+    }
+}
+
+impl Hash for SubscriptionInformation {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.subscriber.hash(state);
+    }
+}
+
+impl Clone for SubscriptionInformation {
```

Comment:
Do we need to explicitly `impl Clone` in this case, since all the members are just having `.clone()` in turn called on them?

Reference [this Rust Playground snippet](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5b7d466181097578531614df33bb3c2d).

Would it suffice to derive `Clone` for `SubscriptionInformation` as shown in the snippet?